File: luhn.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func getChunkSum(chunk string) int {
	sum := 0
	isDouble := false

	for i := len(chunk) - 1; i >= 0; i-- {
		digit, _ := strconv.Atoi(string(chunk[i]))

		if isDouble {
			digit *= 2
			if digit > 9 {
				digit -= 9
			}
		}

		sum += digit
		isDouble = !isDouble
	}

	return sum
}

func main() {
	creditCard := "Credit_Card_Number"
	chunks := strings.Split(creditCard, " ")

	totalSum := 0
	for _, chunk := range chunks {
		totalSum += getChunkSum(chunk)
	}

	fmt.Println(totalSum%10 == 0)
}

File: luhn.py
def get_chunk_sum(chunk):
    total_sum = 0
    is_double = False
    
    for digit in reversed(chunk):
        digit = int(digit)
        
        if is_double:
            digit *= 2
            if digit > 9:
                digit -= 9
        
        total_sum += digit
        is_double = not is_double

    return total_sum

credit_card = "Credit_Card_Number"
chunks = credit_card.split()
total_sum = sum(get_chunk_sum(chunk) for chunk in chunks)

print(total_sum % 10 == 0)
